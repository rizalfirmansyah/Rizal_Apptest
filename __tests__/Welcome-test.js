import React from 'react';
import renderer from 'react-test-renderer';
import Welcomescreens from '../src/screens/welcomescreens';
import { fireEvent, render, waitFor } from '@testing-library/react-native'

test('renders correctly', () => {
    const tree = renderer.create(<Welcomescreens />).toJSON();
    expect(tree).toMatchSnapshot();
});


describe('Welcome screen', () => {

    it('navigate welcomescreen to homescreen', () => {
        const navigation = { navigate: () => { } }
        jest.spyOn(navigation, 'navigate');
        const page = render(<Welcomescreens navigation={navigation} />);
        const nextButton = page.getByTestId('nextButton');
        fireEvent.press(nextButton);
        expect(navigation.navigate).toHaveBeenCalledWith("homescreens");
    })

    

})