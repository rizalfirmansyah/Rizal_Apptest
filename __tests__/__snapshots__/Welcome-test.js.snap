// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "50%",
        "justifyContent": "flex-end",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/welcome1.jpeg",
        }
      }
      style={
        Object {
          "height": "60%",
          "width": "80%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": "35%",
      }
    }
  >
    <View
      style={
        Object {
          "marginLeft": 20,
          "marginRight": 30,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#55b1f8",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Welcome
      </Text>
    </View>
    <View
      style={
        Object {
          "marginLeft": 20,
          "marginRight": 30,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#7190a3",
            "fontSize": 12,
            "textAlign": "justify",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce at nulla fringilla dui consequat luctus. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": "15%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#55b1f8",
          "borderRadius": 5,
          "marginLeft": 20,
          "marginRight": 30,
          "opacity": 1,
          "padding": 15,
        }
      }
      testID="nextButton"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 15,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
